### PR TITLE
Update busybox 1.32.0 to 1.33.2

### PR DESCRIPTION
**Pull request by danielp96**
Upstreaming busybox version from PureBoot.

1.33.x introduces base32 applet, initially disabled to avoid conflict with tpmtotp base32 (it would overwrite the busybox binary through the symlink and brick the system) later reenabled to replace tpmtotp base32. Both have same encode behavior, the busybox version also decodes while the tpmtotp doesn't, also increase in empty space.

Disable busybox base64, completely unused and saves space.

Busybox patch still applicable, moved to new version.

Adding '--remove-destination' to cp calls on the makefile futureproofs the issue of another module overwriting busybox binary when both provide the same filename.

Empty space of each change:
| Version   | Heads     | Pureboot |
|------------|-------------|------------|
| 1.33.2 with tpmtotp base32                           | 820580  | 2460580 |
| 1.33.2 with busybox base32                          | 822628  | 2461604 |
| 1.33.2 with busybox base32 without base64 | 822628  | 2463140 |

### CODE REVIEW

File: config/busybox.config [hunk 1]
```diff
@@ -55,6 +54,7 @@ CONFIG_EXTRA_LDFLAGS=""
 CONFIG_EXTRA_LDLIBS=""
 # CONFIG_USE_PORTABLE_CODE is not set
 CONFIG_STACK_OPTIMIZATION_386=y
+CONFIG_STATIC_LIBGCC=y
```
[the remark this answers is not in the document]
Comment (danielp96) [reply config/busybox.config:57]: Config option was added on 1.33.0, setting it keeps behavior from 1.32

**Comment by tlaurion** [comment]
@danielp96 making some tests at https://app.circleci.com/pipelines/github/tlaurion/heads?branch=busybox_1.33.2_fixes

**Comment by tlaurion** [comment]
@danielp96 Confirmed that https://app.circleci.com/pipelines/github/tlaurion/heads?branch=busybox_1.33.2_fixes and https://app.circleci.com/pipelines/github/tlaurion/heads?branch=busybox_1.33.2 doesn't change anything and that -O2 seems to be passed correctly.

**Comment by tlaurion** [comment]
Rebased on master under https://app.circleci.com/pipelines/github/tlaurion/heads?branch=danielp96_busybox_version_bump
and tested on x230 through x230-hotp-maximized, and base32 works correctly where called to seal-totp.

Works as expected, even though talos II build fails downloading https://mirror.csclub.uwaterloo.ca/gnu/mpc/mpc-1.2.1.tar.gz at coreboot preparation of its cross-compiled buildstack.... Will test on talos II when rom finally built

**Pull request merged**